src/web/extractors.rs

```diff
@@ -729,14 +753,113 @@ impl FromRequest<ServerState> for BsoQueryParams {
                     RequestErrorLocation::QueryString,
                     None,
                 )
-            })?.into_inner();
+            })?
+            .into_inner();
         params.validate().map_err(|e| {
             ValidationErrorKind::FromValidationErrors(e, RequestErrorLocation::QueryString)
         })?;
         Ok(params)
     }
 }
 
+#[derive(Debug, Default, Clone, Deserialize, Validate)]
+#[serde(default)]
+pub struct BatchParams {
+    pub batch: Option<String>,
+    #[validate(custom = "validate_commit")]
```

Comment:
Is this query or header params? I was including a bit in the validate function to indicate where they came from so the error would properly reflect the location. It'd be nice if the validate was capable of passing args as a partial so we could avoid funcs for each location.